modules/crud/schema.go

```diff
@@ -132,7 +137,116 @@ func getFieldType(fieldType ast.Type, fieldTypeStuct *SchemaFieldType, doc *ast.
 	return nil
 }
 
+func (m *Module) schemaValidator(dbType, col string, fields map[string]interface{}) error {
+
+	collectionFields := m.schema[dbType][col]
+	for fieldKey, fieldValue := range collectionFields {
+		// TODO: check if internal elements of list is required for this we need to change schmea parser
+		// check if key is required
+		value, ok := fields[fieldKey]
+		if fieldValue.IsFieldTypeRequired {
+			if !ok {
+				return errors.New("Field " + fieldKey + " Not Present")
+			}
+		}
+
+		// check type
+		fieldType, err := m.checkType(value, fieldValue.Kind, dbType, fieldValue.TableJoin)
+		if err != nil {
+			return err
+		}
+
+		if fieldType != fieldValue.Kind {
+			return errors.New("Wrong Type Wanter " + fieldValue.Kind + " got " + fieldType)
+		}
+
+	}
+
+	return nil
+}
+
 // ValidateSchema checks data type
-func (m *Module) ValidateSchema() {
+func (m *Module) ValidateSchema(dbType, col string, req *model.CreateRequest) error {
+
+	v := make([]map[string]interface{}, 1)
+
+	switch t := req.Document.(type) {
+	case []map[string]interface{}:
+		v = t
+	case map[string]interface{}:
+		v = append(v, t)
+	}
+
+	for _, fields := range v {
+		err := m.schemaValidator(dbType, col, fields)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Module) checkType(value interface{}, kind, dbType, col string) (string, error) {
+
+	switch v := value.(type) {
+	case int:
+		// TODO: int64
+		switch kind {
+		case TypeDateTime:
+			unitTimeInRFC3339 := time.Unix(int64(v), 0).Format(time.RFC3339)
```

Comment:
Do we need to do that `.Format`

Reply:
Weren't you supposed to overwrite the field with a date object?

Reply:
> 
> 
> Do we need to do that `.Format`

func main() {
    theValue := 1405544146
    unixTimeUTC:=time.Unix(theValue, 0) //gives unix time stamp in utc 

    unitTimeInRFC3339 :=unixTimeUTC.Format(time.RFC3339) // converts utc time to RFC3339 format

    fmt.Println("unix time stamp in UTC :--->",unixTimeUTC)
    fmt.Println("unix time stamp in unitTimeInRFC3339 format :->",unitTimeInRFC3339)
}

This is something that u sent that' why I added it
and graphql spec. uses rfc3339

Reply:
> 
> 
> Weren't you supposed to overwrite the field with a date object?

which field u are talking about.
The actual field of type dateTime

Reply:
So any number is mostly a valid date time object. We don't need to format it as a string at all.

In the docs object. We need to update the field corresponding to the date time kind right?

Reply:
Don't format things here. Return a type `time.Time` directly. No need to reassign value